--- a/modules/videoio/CMakeLists.txt	2025-09-05 15:56:31.621122523 +0000
+++ b/modules/videoio/CMakeLists.txt	2025-09-05 15:59:37.619372507 +0000
@@ -139,10 +139,8 @@
   list(APPEND tgts ocv.3rdparty.v4l)
 endif()
 
-if(TARGET ocv.3rdparty.vin_isp)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__USE_VIN_ISP__")
   ocv_target_link_libraries(${the_module} LINK_PRIVATE m isp isp_ini AWIspApi)
-endif()
 
 if(TARGET ocv.3rdparty.openni2)
   list(APPEND videoio_srcs ${CMAKE_CURRENT_LIST_DIR}/src/cap_openni2.cpp)
@@ -305,3 +303,12 @@
     install(FILES "${ffmpeg_dir}/opencv_videoio_ffmpeg${FFMPEG_SUFFIX}.dll" DESTINATION "bin/" COMPONENT libs RENAME "opencv_videoio_ffmpeg${OPENCV_DLLVERSION}${FFMPEG_SUFFIX}.dll")
   endif()
 endif()
+
+target_link_libraries(opencv_videoio
+    PRIVATE
+    -Wl,--start-group
+    AWIspApi
+    isp
+    isp_ini
+    -Wl,--end-group
+)
\ No newline at end of file
